src/transformers/trainer.py

```diff
@@ -1881,11 +1881,34 @@ def train(
 
         if resume_from_checkpoint is not None:
             if not is_sagemaker_mp_enabled() and not self.is_deepspeed_enabled and not self.is_fsdp_enabled:
-                self._load_from_checkpoint(resume_from_checkpoint)
-            # In case of repeating the find_executable_batch_size, set `self._train_batch_size` properly
-            state = TrainerState.load_from_json(os.path.join(resume_from_checkpoint, TRAINER_STATE_NAME))
-            if state.train_batch_size is not None:
-                self._train_batch_size = state.train_batch_size
+                # Try to load the checkpoints until we find one valid or we run out of checkpoints
+                while resume_from_checkpoint is not None:
+                    try:
+                        self._load_from_checkpoint(resume_from_checkpoint)
+                        break
+                    except Exception as e:
+                        warnings.warn(
+                            f"loading model from {resume_from_checkpoint}: {e}. Trying an even older checkpoint.",
+                            RuntimeWarning,
+                        )
```

Comment:
I don't agree with this blanket wipe here. 

I would rather see a situation of:

1. Check if the directory we are doing `resume_from_checkpoint` has multiple checkpoints we can load from
2. If so, **then** we do a `while` loop, going through them all
3. We should only warn upon a failure, and this should be done at the `logging.warn` level so users can be flagged upon it if needed. Users ignore warnings in logs, so it is only there for informational purposes. 
4. If we have failed to load the initial checkpoint, and we have loaded in a later checkpoint, we should do a full `print` to the user, alerting them that the prior/initial checkpoint has failed and we are resuming from X state.